compose/parallel.py

```diff
@@ -68,6 +61,23 @@ def parallel_execute(objects, func, index_func, msg):
             raise error
 
 
+def setup_queue(writer, objects, func, index_func):
+    for obj in objects:
+        writer.initialize(index_func(obj))
+
+    q = Queue()
+
+    # TODO: limit the number of threads #1828
+    for obj in objects:
+        t = Thread(
+            target=perform_operation,
+            args=(func, obj, q.put, index_func(obj)))
+        t.daemon = True
+        t.start()
+
+    return q
```

Comment:
This was extracted to reduce the complexity of `parallel_execute()` to keep it below the current threshold of 11. Adding the extra `except` brought it to 12.